### ISSUE TITLE
TES IV: Oblivion has broken water

### ISSUE THREAD

**Issue by xddxd** [RPCS3/rpcs3#9033]
When looking at the water in Oblivion BLUS30087 at a specific angle, the reflections completely break.
RRC: https://www.dropbox.com/s/p8fji8hdzhmjlr0/captures.zip?dl=0
![Screenshot_11](https://user-images.githubusercontent.com/1712979/95023407-911dd300-0685-11eb-90fd-324f253c061e.png)
Log: [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/5323978/RPCS3.log.gz)

R7 2700
RX 580 20.9.2
Windows 10 20H2

**Comment by Thor-x86** [comment]
Is it also break when using OpenGL?

**Comment by maximumxp** [comment]
@Thor-x86 
Broken when using OpenGL, too
![image](https://user-images.githubusercontent.com/19251031/100520916-48633180-317f-11eb-86f7-ae664798a410.png)

**Comment by Vembumees** [comment]
just tried oblivion, had same thing when resolution scale was above 100%, set it to default then it was fine.

**Comment by Vembumees** [comment]
got this type of glitches and other (like parts of ui disappearing) when I played above native resolution, at normal 100% res its fine.

**Comment by maximumxp** [comment]
As someone correctly told me, this happens only when the resolution is NOT native.

**Comment by maximumxp** [comment]
After some testing, upping the resolution to 1920x1080 and putting the Resolution Scale Treshold to 135x135 worked fine on Vulkan.

**Comment by Vembumees** [comment]
I said it before, but deleted, because it happened even with native. Played the game for a while, seems the graphics sometimes just gets glitchy (only have played on vulkan), especially visible before. Also game sometimes starts freezing. Usually quiting to the menu (just reloading doesnt fix them) and returning to the game fixes the visual glitches, I can only guess some cache stuff gets full somewhere and causes these problems.

**Comment by kd-11** [comment]
Any change with current rpcs3?

**Comment by Darkhost1999** [comment]
This issue is very rare on default settings. Nearly impossible to encounter.
With WCB enabled it was still rare to encounter. Although not as rare.
With all the Read and Write buffers enabled like how the only log in this chat thread has enabled. This broken water effect can appear entirely at random with precise camera angles. Anywhere from extremely rare to very frequent.
I attempted 3 times to get an RSX capture.
Attempt 1: Black screen.
Attempt 2: Game froze while trying to get to the water.
Attempt 3: The issue did not show up for more than 10 minutes of walking along the bank of a river. Even with all buffers enabled.

Tldr: This issue is nearly impossible to encounter at default RPCS3 settings on latest master.

**Comment by kd-11** [comment]
Time for a retest. Might be fixed.
If its still an issue, maybe we can try to find some way to reproduce the bug deterministically so it can be investigated further.

**Comment by xddxd** [comment]
![image](https://user-images.githubusercontent.com/1712979/151852654-1d3add6e-52fe-4dd1-9ead-09c218487ac3.png)
Water is completely gone now.

**Comment by Darkhost1999** [comment]
* All RSX buffers 300% resolution scale
![image](https://user-images.githubusercontent.com/60384196/152900123-6cbf7489-b5ce-485f-afb1-d484d7741515.png)
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/8019885/RPCS3.log)
* All RSX buffers 250% resolution scale
![image](https://user-images.githubusercontent.com/60384196/152900373-f0a1c5af-c527-412a-b7db-9f9cfc47f812.png)
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/8019896/RPCS3.log)
* All RSX buffers 200% resolution scale
![image](https://user-images.githubusercontent.com/60384196/152900550-d476bd83-495d-4a14-b01c-77b655321dd8.png)
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/8019909/RPCS3.log)
* All RSX buffers 150% resolution scale
![image](https://user-images.githubusercontent.com/60384196/152900951-0819f04d-7bd6-45a5-890d-9e3f3f87f253.png)
oops forgot log right here
* No RSX buffers 150% resolution scale
![image](https://user-images.githubusercontent.com/60384196/152901091-7bcbe52a-6906-4c67-8ff3-c3477b7fdda0.png)
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/8019939/RPCS3.log)

**Comment by Darkhost1999** [comment]
OpenGl for good measure
* No RSX buffers 300% resolution scale
![image](https://user-images.githubusercontent.com/60384196/152901727-ddb09813-ff85-474f-b532-e25dceffa68f.png)
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/8019961/RPCS3.log)

* No RSX buffers 250% resolution scale
![image](https://user-images.githubusercontent.com/60384196/152901961-eb22063e-d5b3-4017-8db9-75ecbfd85751.png)
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/8019991/RPCS3.log)

**Comment by Darkhost1999** [comment]
Because upscaling is a hack. And the issue is only present when upscaling to values not equal to 100, 200, 300, x00%.
Is this even an RPCS3 issue anymore?

**Comment by Darkhost1999** [comment]
Issues like the OP posted do not happen anymore. Only issue is when using resolution scaling. Water just works

**Comment by kd-11** [comment]
@xddxd Please retest.

**Comment by kd-11** [comment]
Closing due to inactivity.
